qdrant/v1.2.x/search.md

```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
 However, using the offset parameter saves the resources by reducing network traffic and the number of times the storage is accessed.
 
 Using an `offset` parameter, will require to internally retrieve `offset + limit` points, but only access payload and vector from the storage those points which are going to be actually returned.
+
+## Grouping API
+
+*Available since v1.2.0*
+
+It is possible to group results by a certain field. This is useful when you have multiple points for the same item, and you want to avoid redundancy of the same item in the results.
+
+For example, if you have a large document split into multiple chunks, and you want to search or recommend on a per-document basis, you can group the results by the document ID. 
+
+Consider having points with the following payloads:
+
+```json
+{
+    {
+        "id": 0,
+        "payload": {
+            "chunk_part": 0, 
+            "document_id": "a",
+        },
+        "vector": ...,
+    },
+    {
+        "id": 1,
+        "payload": {
+            "chunk_part": 1, 
+            "document_id": "a",
+        },
+        "vector": ...,
+    },
+    {
+        "id": 2,
+        "payload": {
+            "chunk_part": 2, 
+            "document_id": "a",
+        },
+        "vector": ...,
+    },
+    {
+        "id": 3,
+        "payload": {
+            "chunk_part": 0, 
+            "document_id": 123,
+        },
+        "vector": ...,
+    },
+    {
+        "id": 4,
+        "payload": {
+            "chunk_part": 1, 
+            "document_id": 123,
+        },
+        "vector": ...,
+    },
+    {
+        "id": 5,
+        "payload": {
+            "chunk_part": 0, 
+            "document_id": 2.42,
+        },
+        "vector": ...,
+    },
+}
+```
+
+With the ***groups*** API, you will be able to get the top *N* points for each document, assuming that the payload of the points contains the document ID.
```

Comment:
Are we returning only full groups OR can we respond with groups with less then N elements?

Reply:
I just realized you have an example afterwards with a group that is not full :+1: 

Wondering if it is worth it to more precise on the grouping semantic.